test/functional/test-3p-nameframe.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getDefaultBootstrapBaseUrl} from '../../src/3p-frame';
+import {createIframePromise} from '../../testing/iframe';
+
+function expectNoContent(win, done) {
+  win.addEventListener('message', content => {
+    expect(content.data).to.have.property('type');
+    expect(content.data.type).to.equal('no-content');
+    done();
+  });
+}
+
+describes.sandboxed('nameframe', {}, () => {
+  let fixture;
+  let win;
+  let doc;
+  let iframe;
+  beforeEach(() => {
+    return createIframePromise(/* runtimeOff */ true).then(f => {
+      fixture = f;
+      win = fixture.win;
+      doc = fixture.doc;
+      iframe = doc.createElement('iframe');
+      iframe.src = getDefaultBootstrapBaseUrl(win, 'nameframe');
```

Comment:
you can just hardcode the URL here. no need for this test to depend on 3p-iframe.js, which is covered with a separate unit test.